### PR TITLE
[Driver][SYCL][NFC] Cleanup various code entries

### PR DESCRIPTION
**Pull request by mdtoguchi**
As we are working towards upstreaming the SYCL support in the compiler, a few areas need to be cleaned up for readability and improved logic to reduce some repetitious output.  Does not impact behaviors in general.

### CODE REVIEW

File: clang/lib/Driver/ToolChains/Clang.cpp [hunk 1]
```diff
@@ -6725,16 +6722,17 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
   // the command line.  This is to allow for CMake based builds using the
   // Linux based driver on Windows to correctly pull in the expected debug
   // library.
-  if (Args.hasArg(options::OPT_fsycl) && !Args.hasArg(options::OPT_nolibsycl)) {
-    if (!D.IsCLMode() && TC.getTriple().isWindowsMSVCEnvironment()) {
+  if (Args.hasArg(options::OPT_fsycl) && !Args.hasArg(options::OPT_nolibsycl) &&
+      !D.IsCLMode()) {
```
Comment (srividya-sundaram) [inline clang/lib/Driver/ToolChains/Clang.cpp:6726]: What difference does it make by combining just `!D.IsCLMode()` in this `if` statement ?
Why not both `!D.IsCLMode() && TC.getTriple().isWindowsMSVCEnvironment()` ?

Comment (mdtoguchi) [reply clang/lib/Driver/ToolChains/Clang.cpp:6726]: This is due to the difference in behavior needed for line#6727 and line#6735.

File: clang/lib/Driver/Driver.cpp [hunk 2]
```diff
@@ -841,17 +841,16 @@ static bool addSYCLDefaultTriple(Compilation &C,
...
   for (const auto &SYCLTriple : SYCLTriples) {
-    if (SYCLTriple.getSubArch() == llvm::Triple::NoSubArch &&
-        SYCLTriple.isSPIROrSPIRV())
+    if (SYCLTriple == DefaultTriple)
       return false;
     // If we encounter a known non-spir* target, do not add the default triple.
     if (SYCLTriple.isNVPTX() || SYCLTriple.isAMDGCN())
       return false;
   }
   // Add the default triple as it was not found.
```
Comment (srividya-sundaram) [inline clang/lib/Driver/Driver.cpp:853]: Is this comment needed anymore?

Comment (mdtoguchi) [reply clang/lib/Driver/Driver.cpp:853]: Yes, the `DefaultTriple` is still being added at line#855

**Review by srividya-sundaram — APPROVED** [review]

**Review by hchilama — APPROVED** [review]

**Review by smanna12 — APPROVED** [review]

**Comment by mdtoguchi** [comment]
@intel/llvm-gatekeepers, this looks to be ready for merge - please take a look, thanks!

**Pull request merged**